src/agent.rs

```diff
@@ -48,14 +48,22 @@ pub enum DecisionRule {
 /// An agent in the simulation
 #[derive(Debug, Deserialize, PartialEq, Clone)]
 pub struct Agent {
+    /// the acronym for the agent.
     pub id: Rc<str>,
+    /// a text description of the agent.
     pub description: String,
+    /// the commodity that the agent produces (could be a service demand too).
     pub commodity_id: String,
     #[serde(deserialize_with = "deserialise_proportion_nonzero")]
+    /// the percentage of the commodity production that the agent is responsible for.
```

Comment:
It's actually between 0 and 1, not a percentage:

```suggestion
    /// The proportion of the commodity production that the agent is responsible for.
```